Gitpod extension authentication docs (#4004)
gitpod-io/website: src/content/docs/references/ides-and-editors/vscode.md

`youtube: kI6firDA0Bw`

## Prerequisites

> To develop in a workspace directly in VS Code desktop, you must install and sign into the Gitpod extension. Use the Visual Studio Code Marketplace to install the [Gitpod extension](https://marketplace.visualstudio.com/items?itemName=gitpod.gitpod-desktop).

Reviewer: in a Gitpod workspace

Reviewer 2: It would be better to explain what the user is expected to do, not what service is used (IMO ok to omit references to the Marketplace).

```suggestion
To open a Workspace directly from VS Code desktop, you'll need to install and sign into the [Gitpod extension](https://marketplace.visualstudio.com/items?itemName=gitpod.gitpod-desktop), which you can find by searching for `Gitpod` inside of VS Code's Extensions tab.
```

> 1. In VS Code, in the Activity Bar, click the Gitpod icon and then click `Sign in`, alternatively, access the Command Palette, for example, by pressing <kbd>Shift+Command+P</kbd> (Mac) / <kbd>Ctrl+Shift+P</kbd> (Windows/Linux) and type **gitpod**, click `Gitpod: Sign In`

Reviewer 2: Could we use the Keybind component like we do here? https://github.com/gitpod-io/website/blob/e583ce62afc8e887e0f68b30ebf5e95cb9201b8b/src/content/changelog/2021-09.md?plain=1#L19

```suggestion
1. In VS Code, from the Activity Bar, click the Gitpod icon and then choose `Sign in`. Alternatively, open the Command Palette <kbd>Shift+Command+P</kbd> (Mac) / <kbd>Ctrl+Shift+P</kbd> (Windows/Linux) and type **gitpod**. Then select `Gitpod: Sign In`
```

![Sign in VS Code from Gitpod view](/images/editors/vscode-desktop-view-signin.png)

> _Sign in VS Code Desktop from the Gitpod view_
> 
> ![Sign in VS Code from Command Palette](/images/editors/vscode-desktop-command-signin.png)
> _Sign in VS Code Desktop from the Command Palette_

Reviewer 2: ```suggestion
_Sign in to VS Code Desktop from the Gitpod view_

![Sign in VS Code from Command Palette](/images/editors/vscode-desktop-command-signin.png)
_Sign in to VS Code Desktop from the Command Palette_
```

> 2. A modal dialog will show up prompting to open an external website in your browser, click `Open`.

Reviewer: It is too technical, better something what explain in user terms and why it happens:

A dialog will appear asking permission to open your browser for authenticating VS Code with Gitpod. Click `Open` to proceed.

Author: ```suggestion
2. A dialog will appear asking permission to open your browser for authenticating VS Code with Gitpod. Click Open to proceed.
```

![Open external website](/images/editors/vscode-desktop-open-external.png)

> 3. Check your browser, you should see a Gitpod tab asking to authorize the VSCode Gitpod extension, click `Authorize`.

Reviewer 2: Could we make this simpler? Maybe we're overcomplicating this pretty straight forward step.

```suggestion
3. Authorize the extension from within the newly opened browser tab.
```

Reviewer: It is important point out here if a browser is not opened automatically then a user has to check. It did happen before.

Reviewer 2: Maybe add a troubleshooting section to the bottom? I don't think we should document all edge cases in the instructions @akosyakov

Reviewer: We are adding the instruction because of such things. I think it is alright. We are going to point to this section from VS Code when Sign in is failing.

![Authorize VSCode Gitpod extension](/images/editors/vscode-desktop-authorize.png)

> 4. You will be redirected to VSCode and authentication should be completed. In the Activity Bar, click the Accounts icon and you should see your Gitpod account listed.

Reviewer: VS Code

Author: ```suggestion
4. You will be redirected to VS Code and authentication should be completed. In the Activity Bar, click the Accounts icon and you should see your Gitpod account listed.
```

> ![VSCode Gitpod extension authenticated](/images/editors/vscode-desktop-authenticated.png)

Reviewer 2: ```suggestion
![Authenticated VS Code Gitpod extension](/images/editors/vscode-desktop-authenticated.png)
```
